### ISSUE TITLE
Unable to yaml.safe_load() a second time after enabling pyjion

### ISSUE THREAD

**Issue by FranzForstmayr** [tonybaloney/Pyjion#309]
Hi, I tried to use Pyjion the first time and installed pyjion==1.0.0a on my Ubuntu 20.04 machine.
I've dotnet preview 6.0 installed and pyjion seems to run pretty good the first time.

However I notized that I'm unable to read the same yaml string more than once after import pyjion.
See the following reproducer:

```python
import yaml

content = """
linear: 
  - [1, 2, 3]
  - test
  - 1
"""

def read_yaml():
    return yaml.safe_load(content)

print("This will work")
print(read_yaml())

print("\nSecond time too")
print(read_yaml())

import pyjion
pyjion.enable()

print("\nFirst time after pyjion enabled works as well")
print(read_yaml())

print("\nSecond time not not")
print(read_yaml())

print("\nVersion", pyjion.__version__)
```
This will result in the following stdout:

```
This will work
{'linear': [[1, 2, 3], 'test', 1]}

Second time too
{'linear': [[1, 2, 3], 'test', 1]}

First time after pyjion enabled works as well
{'linear': [[1, 2, 3], 'test', 1]}

Second time not
{'linear': ['test', 1]}

Version 1.0.0a1
```

I'm still able to get objects like integers or strings, but not the nested list object.

**Comment by FranzForstmayr** [comment]
I just found out, that 1.0.0a1 is not the newest release, so I tried 1.0.0b4 as well. 
However, the issue remains,

**Comment by tonybaloney** [comment]
thanks for the report @FranzForstmayr this makes a great test case.

The second-run will be optimizing based on observations on the first run. Something is over-optimising and creating an invalid JIT path.

As a workaround you can disable PGC,

`pyjion.disable_pgc()` 

That _should_ be a temporary fix, but I will write a test for this and figure out where the problem lies.

**Comment by FranzForstmayr** [comment]
Great! 
Thank you for your quick response, I'm happy to help where I can.

**Comment by tonybaloney** [comment]
Fixed in 1.1.1
